src/WebJobs.Extensions.IoTHub/IoTHubConfigProvider.cs

```diff
@@ -0,0 +1,130 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Text;
+using Microsoft.Azure.Devices;
+using Microsoft.Azure.WebJobs.Host.Bindings;
+using Microsoft.Azure.WebJobs.Host.Config;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Azure.WebJobs.Extensions.IoTHub
+{
+    /// <summary>
+    /// Binding provider for IoTHub
+    /// </summary>
+    public class IoTHubConfigProvider : IExtensionConfigProvider,
+        IConverter<byte[], Message>,
+        IConverter<string, Message>,
+        IConverter<JObject, Message>,
+        IConverter<IoTHubAttribute, ServiceClient>
+    {
+        internal const string IoTHubConnectionStringName = "AzureWebJobsIoTHub";
```

Comment:
line 38, `nameResolver.Resolve(IoTHubConnectionStringName)`, as well as in the `ValidateConnection` error message